ui/src/app/pages/access/access-renewal-page.spec.tsx

```diff
@@ -0,0 +1,169 @@
+      profileStore.set({profile: profileStore.get().profile, load, reload, updateCache});
+    });
+
+    profileStore.set({profile, load, reload, updateCache});
+
+    serverConfigStore.set({config: {
+      enableDataUseAgreement: true,
+      gsuiteDomain: 'fake-research-aou.org',
+      projectId: 'aaa',
+      publicApiKeyForErrorReports: 'aaa',
+      enableEraCommons: true,
+      enableComplianceTraining: true
+    }});
+
+    const institutionApi = new InstitutionApiStub();
+    registerApiClient(InstitutionApi, institutionApi);
+  });
+
+
+  it('should render when the user is expired', async () => {
+    expireAllModules()
+    const wrapper = component();
+
+    setCompletionTimes(() => Date.now() - 1000 * 60 * 60 * 24 * EXPIRY_DAYS);
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(findNodesContainingText(wrapper, 'access has expired').length).toBe(1);
+    expect(findNodesByExactText(wrapper, 'Review').length).toBe(1)
+    expect(findNodesByExactText(wrapper, 'Confirm').length).toBe(1)
+    expect(findNodesByExactText(wrapper, 'View & Sign').length).toBe(1)
+    expect(findNodesByExactText(wrapper, 'Complete Training').length).toBe(1);
+  });
+
+  it('should show the correct state when profile is complete', async () => {
+    expireAllModules()
+
+    const wrapper = component();
+
+    udpateOneModule('profileConfirmation', oneYearFromNow());
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(findNodesByExactText(wrapper, 'Confirmed').length).toBe(1)
+    expect(findNodesByExactText(wrapper, 'Confirm').length).toBe(1);
+    expect(findNodesByExactText(wrapper, 'View & Sign').length).toBe(1)
+    expect(findNodesByExactText(wrapper, 'Complete Training').length).toBe(1);
+  });
+
+  it('should show the correct state when profile and publication are complete', async () => {
+    expireAllModules()
+
+    const wrapper = component();
+
+    udpateOneModule('profileConfirmation', oneYearFromNow());
+    udpateOneModule('publicationConfirmation', oneYearFromNow());
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(findNodesByExactText(wrapper, 'Confirmed').length).toBe(2);
+    expect(findNodesByExactText(wrapper, 'View & Sign').length).toBe(1)
+    expect(findNodesByExactText(wrapper, 'Complete Training').length).toBe(1);
+  });
+
+  it('should show the correct state when all items except DUCC are complete', async () => {
+    expireAllModules()
+
+    const wrapper = component();
+
+    udpateOneModule('profileConfirmation', oneYearFromNow());
+    udpateOneModule('publicationConfirmation', oneYearFromNow());
+    udpateOneModule('complianceTraining', oneYearFromNow());
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(findNodesByExactText(wrapper, 'Confirmed').length).toBe(2);
+    expect(findNodesByExactText(wrapper, 'View & Sign').length).toBe(1);
+    expect(findNodesByExactText(wrapper, 'Completed').length).toBe(1);
+  });
+
+  it('should show the correct state when all items are complete', async () => {
```

Comment:
these are nice tests, thanks.  Could you add one for Bypassed modules?

Reply:
will do in another update